loader/extension_manual.c

```diff
@@ -105,20 +105,16 @@ VKAPI_ATTR VkResult VKAPI_CALL terminator_GetPhysicalDeviceSurfaceCapabilities2E
     struct loader_physical_device_term *phys_dev_term = (struct loader_physical_device_term *)physicalDevice;
     struct loader_icd_term *icd_term = phys_dev_term->this_icd_term;
 
-    VkIcdSurface *icd_surface = (VkIcdSurface *)(uintptr_t)(surface);
-
-    // Unwrap the surface if needed
-    VkSurfaceKHR unwrapped_surface = surface;
-    if (NULL != phys_dev_term->this_icd_term->surface_list.list &&
-        phys_dev_term->this_icd_term->surface_list.capacity > icd_surface->surface_index * sizeof(VkSurfaceKHR) &&
-        phys_dev_term->this_icd_term->surface_list.list[icd_surface->surface_index]) {
-        unwrapped_surface = phys_dev_term->this_icd_term->surface_list.list[icd_surface->surface_index];
+    if (VK_NULL_HANDLE != surface) {
```

Comment:
`VUID-vkGetPhysicalDeviceSurfaceCapabilities2EXT-surface-parameter` to be pedantic, vkGetPhysicalDeviceSurfaceCapabilities2EXT is not one of the functions that VK_GOOGLE_surfaceless_query allows to be surfaceless. 

Frankly, I think VK_GOOGLE_surfaceless_query is poorly written for that reason (some query functions have it, others dont?) I guess the list is what android uses since thats their target, but this loader is occasionally used in mobile-esq situations so should respect the extension.

Reply:
> Frankly, I think VK_GOOGLE_surfaceless_query is poorly written for that reason (some query functions have it, others dont?)

Yes, this is what confused me around this too.

I've fixed this.